sources/engine/Stride.Graphics/VertexPositionNormalTextureSkinning.cs

```diff
@@ -0,0 +1,130 @@
+    public struct VertexPositionNormalTextureSkinning : IEquatable<VertexPositionNormalTextureSkinning>, IVertex
+    {
+        /// <summary>
+        /// Initializes a new <see cref="VertexPositionNormalTextureSkinning"/> instance.
+        /// </summary>
+        /// <param name="position">The position of this vertex.</param>
+        /// <param name="normal">The normal of this vertex.</param>
+        /// <param name="textureCoordinate">UV texture coordinates.</param>
+        /// <param name="blendIndices">The blend indices (joints) of this vertex.</param>
+        /// <param name="blendWeights">The blend weights (joints) of this vertex.</param>
+        public VertexPositionNormalTextureSkinning(Vector3 position, Vector3 normal, Vector2 textureCoordinate, Int4 joint, Vector4 weight)
+            : this()
+        {
+            Position = position;
+            Normal = normal;
+            TextureCoordinate = textureCoordinate;
+            BlendIndices = joint;
+            BlendWeight = weight;
+        }
+
+        /// <summary>
+        /// XYZ position.
+        /// </summary>
+        public Vector3 Position;
+
+        /// <summary>
+        /// Normal vector.
+        /// </summary>
+        public Vector3 Normal;
+
+        /// <summary>
+        /// UV texture coordinates.
+        /// </summary>
+        public Vector2 TextureCoordinate;
+
+        /// <summary>
+        /// BlendIndices vector.
+        /// </summary>
+        public Int4 BlendIndices;
+
+        /// <summary>
+        /// BlendWeight vector.
+        /// </summary>
+        public Vector4 BlendWeight;
+
+
+
+        /// <summary>
+        /// Defines structure byte size.
+        /// </summary>
+        public static readonly int Size = 12 + 12 + 8 + 4 + 16;
+
+        /// <summary>
+        /// The vertex layout of this struct.
+        /// </summary>
+        public static readonly VertexDeclaration Layout =
+            new VertexDeclaration(
+                VertexElement.Position<Vector3>(),
+                VertexElement.Normal<Vector3>(),
+                VertexElement.TextureCoordinate<Vector2>(),
+                new VertexElement(VertexElementUsage.BlendIndices, 3, PixelFormat.R32G32B32A32_UInt, VertexElement.AppendAligned),
+                new VertexElement(VertexElementUsage.BlendWeight, 4, PixelFormat.R32G32B32A32_Float, VertexElement.AppendAligned)
+            );
+
+        public bool Equals(VertexPositionNormalTextureSkinning other)
+        {
+            return
+                Position.Equals(other.Position) &&
+                Normal.Equals(other.Normal) &&
+                TextureCoordinate.Equals(other.TextureCoordinate) &&
+                BlendIndices.Equals(other.BlendIndices) &&
+                BlendWeight.Equals(other.BlendWeight);
+        }
+
+        public override bool Equals(object obj)
+        {
+            if (obj is null) return false;
+            return obj is VertexPositionNormalTextureSkinning && this.Equals(obj);
```

Comment:
``this.Equals(obj)`` recurses upon itself indefinitely afaik since obj is still of type object, use this instead:
```cs
public override bool Equals(object obj) => obj is VertexPositionNormalTextureSkinning vpnts && this.Equals(vpnts);
```